### PR TITLE
pkgs/servers: switch to fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
use fetchFromGitHub

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/apache-modules/mod_fastcgi/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, apacheHttpd }:
+{ lib, stdenv, fetchFromGitHub, apacheHttpd }:
```
Comment (Mic92) [inline pkgs/servers/http/apache-modules/mod_fastcgi/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, fetchurl, apacheHttpd }:
```

Comment (Mic92) [reply pkgs/servers/http/apache-modules/mod_fastcgi/default.nix:1]: still required according to evaluation.

Comment (Stunkymonkey) [reply pkgs/servers/http/apache-modules/mod_fastcgi/default.nix]: is fixed

**Review by zimbatm — APPROVED** [review]

**Comment by risicle** [comment]
On nixos x86_64

```
builder for '/nix/store/j7ixx9ll3r59cxs0v7mbhqjhv6xc3bl2-mod_fastcgi-2.4.7.1.drv' failed with exit code 1; last 10 log lines:
  unpacking source archive /nix/store/p4ag3062fia843q6sla2rmbpy0n2nbzl-source
  source root is source
  patching sources
  applying patch /nix/store/pw15rd1k2jjh08sy4945xvh3mzwniljd-compile-against-apache24.diff
  patching file fcgi.h
  Hunk #2 FAILED at 61.
  Hunk #3 FAILED at 369.
  2 out of 3 hunks FAILED -- saving rejects to file fcgi.h.rej
  patching file mod_fastcgi.c
  Hunk #2 succeeded at 3017 (offset -1 lines).
```

**Comment by Stunkymonkey** [comment]
any idea why it did not fail before? for me the code looks identical.

i can easily remove the commit.

**Comment by risicle** [comment]
Well, can anyone reproduce my failure?

**Comment by Stunkymonkey** [comment]
i can reproduce it as well.

**Comment by risicle** [comment]
Could upstream have changed the tagged revision and we've all just been relying on the cache.nixos.org version since then?

**Comment by zimbatm** [comment]
why do the hashes change by the way? I would expect the URLs to be the same.

**Comment by zimbatm** [comment]
One possible reason for the patch apply failure is that a previous version bump didn't update the hash so it was still pulling the old source from the cache.

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 138526` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mod_fastcgi (apacheHttpdPackages.mod_fastcgi)</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>mod_wsgi (apacheHttpdPackages.mod_wsgi)</li>
    <li>apacheHttpdPackages.mod_wsgi3 (apacheHttpdPackages_2_4.mod_wsgi3)</li>
    <li>beanstalkd</li>
    <li>couchpotato</li>
    <li>fcgiwrap</li>
    <li>slimserver</li>
    <li>ursadb</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> One possible reason for the patch apply failure is that a previous version bump didn't update the hash so it was still pulling the old source from the cache.

This is exactly what happened https://github.com/Stunkymonkey/nixpkgs/commit/0356c1b87d2d523bfd69b59cfc129dda77c77bec#diff-e68f5d809d2fa238184225663fa070248a2d805c320ddeab932e4e7c45b9c33fR17

Since no one noticed it since over 2 years and the project seems to be dead anyway I would suggest to mark it broken and let someone interested in the project deal with it.

**Comment by Stunkymonkey** [comment]
should this be a separate PR? or is it fine like this?

**Pull request merged**